telegram/bot.py

```diff
@@ -1222,10 +1244,14 @@ def getUpdates(self, offset=None, limit=100, timeout=0, network_delay=5., **kwar
 
         if result:
             self.logger.debug('Getting updates: %s', [u['update_id'] for u in result])
+
         else:
             self.logger.debug('No new updates found.')
 
-        return [Update.de_json(x) for x in result]
+        for u in result:
+            self.create_references(u)
+
+        return [Update.de_json(u) for u in result]
 
```

Comment:
@jh0ker 
I'm conflicted here.
I know it's pretty the way you currently implemented it (explicit and self explanatory). However, on busy bots, you actually run twice on 100 updates each time this code is called. so it can actually have performance impact.

Suggestion for discussion:
Instead of having `create_references` method, the various `de_json` methods will accept `bot` as a second parameter (similar to what I've done with `File.de_json` for the `request` object) 

then you can do:
`return [Update.de_json(u, self) for u in result]`

A similar comment goes for the Webhook below